scripts/page_browse.js

```diff
@@ -0,0 +1,212 @@
+/* global $ pageBrowserData dpData initSplit */
+/* exported hiddenProject pageChanger splitter imageControl textControl roundSelect */
+
+function attrSafe(str) {
+    return str.replace("'", "&apos;");
+}
+
+var hiddenProject = function() {
+    return $("<input>", {type: 'hidden', name: 'project', value: pageBrowserData.projectid});
+};
+
+var pageChanger = function () {
+    let pageSelector = document.createElement("select");
+    pageSelector.name = 'imagefile';
+    let prevButton = $("<input>", {type: 'submit', value: attrSafe(dpData.strings.previous)});
+    let nextButton = $("<input>", {type: 'submit', value: attrSafe(dpData.strings.next)});
+
+    pageBrowserData.pages.forEach(function(page) {
+        let selected = pageBrowserData.currentPage === page;
+        pageSelector.add(new Option(page, page, selected, selected));
+    });
+
+    $(pageSelector).change(function() {
+        this.form.submit();
+    });
+
+    prevButton.prop("disabled", pageSelector.selectedIndex === 0);
+    nextButton.prop("disabled", pageSelector.selectedIndex === (pageSelector.length - 1));
+
+    prevButton.click(function () {
+        pageSelector.selectedIndex -= 1;
+    });
+
+    nextButton.click(function () {
+        pageSelector.selectedIndex += 1;
+    });
+    return [dpData.strings.page, ": ", pageSelector, prevButton, nextButton];
+};
+
+
+var splitter = function(container) {
+    const textImageSplitID = "text_image_split";
+    const splitPercentID = "split_percent";
+    // stored value is "H" or "V", vSplit is true or false
+    // if not set vSplit will be false
+    let vSplit = ("V" === localStorage.getItem(textImageSplitID));
+    var splitPercent = localStorage.getItem(splitPercentID);
+    if(!splitPercent) {
+        splitPercent = 50;
+    }
+    var mainSplit;
+
+    let vSwitchButton = $("<button>", {type: 'button', class: 'img-button', title: attrSafe(dpData.strings.switchVert)}).append($("<img>", {src: dpData.buttonImages.imgVSplit}));
+    let hSwitchButton = $("<button>", {type: 'button', class: 'img-button', title: attrSafe(dpData.strings.switchHoriz)}).append($("<img>", {src: dpData.buttonImages.imgHSplit}));
+
+    function setSplitControls(vSplit) {
+        if (vSplit) {
+            hSwitchButton.show();
+            vSwitchButton.hide();
+        } else {
+            hSwitchButton.hide();
+            vSwitchButton.show();
+        }
+    }
+
+    function changeSplit(vSplit) {
+        mainSplit.setSplit(vSplit);
+        setSplitControls(vSplit);
+        localStorage.setItem(textImageSplitID, vSplit ? "V" : "H");
+    }
+
+    vSwitchButton.click(function () {
+        changeSplit(true);
+    });
+
+    hSwitchButton.click(function () {
+        changeSplit(false);
+    });
+
+    mainSplit = initSplit({paneContainer: container, verticalSplit: vSplit, splitPercent: splitPercent});
+    setSplitControls(vSplit);
+
+    $(window).resize(mainSplit.reLayout);
```

Comment:
The splitter should register this event callback itself.

Reply:
> Focused this review primarily on the splitter.

Thanks Casey.

What we really want to do is to re-layout the splitter when the contaner resizes. This would cover both the case of the window resizing and for a subsidiary splitter (like the one in the mentors page) needing to be re-layed out when the splitter bar in the containing splitter is moved (which is now done by a callback). It would be possible to do this using https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver. But this is not implemented in IE. I'll try coding it.

Reply:
Changed to adding another parameter for relayout. Not sure if it's an improvement.